pkg/kine/drivers/generic/generic.go

```diff
@@ -329,16 +342,26 @@ func (d *Generic) query(ctx context.Context, txName, sql string, args ...interfa
 			time.Sleep(jitter.Deviation(nil, 0.3)(2 * time.Millisecond))
 			continue
 		}
+		done()
```

Comment:
I am not sure I get this. Why do we call `done()` on every iteration?

Reply:
This part of the code tries to query the DB. If it fails it will retry in a loop 500 times. So, this `done` call will only be reached once the DB query finishes without error (otherwise the loop will continue due to the `continue` call on line 343.

To cover both cases, successful DB query and failed query after 500 tries, we need to call the `done` method once in the loop and once on line 349.

Reply:
Ah, I see now, thanks